Failure in `ensure_plottable_histogram`?

Hi, since uhi v1.1.0 the `histoprint` tests fail:

https://github.com/scikit-hep/histoprint/actions/runs/24653077923/job/72080015205#step:6:306

```
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/uhi/numpy_plottable.py:390: in ensure_plottable_histogram
    return NumPyPlottableHistogram(*(np.asarray(h) for h in hist))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        hist       = ([array([  7,  44, 108, 197, 263, 214, 115,  34,  14,   4,   0,   0,   0,
         0,   0]), array([  0,   0,   0,   3... 0.33333333,  1.        ,
        1.66666667,  2.33333333,  3.        ,  3.66666667,  4.33333333,
        5.        ]))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[AttributeError("'NumPyPlottableHistogram' object has no attribute 'axes'") raised in repr()] NumPyPlottableHistogram object at 0x7f139739c4d0>
hist = array([[  7,  44, 108, 197, 263, 214, 115,  34,  14,   4,   0,   0,   0,
          0,   0],
       [  0,   0,   0,   3...5,  96,
         45,   7],
       [  4,   9,  22,  29,  39,  59,  60,  74,  54,  47,  41,  23,  15,
         15,   3]])
variances = None, kind = <Kind.COUNT: 'COUNT'>
bins = (array([-5.        , -4.33333333, -3.66666667, -3.        , -2.33333333,
       -1.66666667, -1.        , -0.33333333,  0.33333333,  1.        ,
        1.66666667,  2.33333333,  3.        ,  3.66666667,  4.33333333,
        5.        ]),)
raw_bins = (array([-5.        , -4.33333333, -3.66666667, -3.        , -2.33333333,
       -1.66666667, -1.        , -0.33333333,  0.33333333,  1.        ,
        1.66666667,  2.33333333,  3.        ,  3.66666667,  4.33333333,
        5.        ]),)

    def __init__(
        self,
        hist: np.typing.NDArray[Any],
        *bins: (
            np.typing.NDArray[Any] | None | tuple[np.typing.NDArray[Any] | None, ...]
        ),
        variances: np.typing.NDArray[Any] | None = None,
        kind: Kind = Kind.COUNT,
    ) -> None:
        self._values = hist
        self._variances = variances
    
        if len(bins) == 1 and isinstance(bins[0], tuple):
            (bins,) = bins  # type: ignore[assignment]
    
        if len(bins) == 0:
            bins = tuple([None] * len(hist.shape))
    
        raw_bins: tuple[np.typing.NDArray[Any] | None, ...] = bins  # type: ignore[assignment]
    
        self.kind = kind
        self.axes: Sequence[PlottableAxis] = [
>           _bin_helper(shape, b) for shape, b in zip(hist.shape, raw_bins, strict=True)
                                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        ]
E       ValueError: zip() argument 2 is shorter than argument 1
```

Was there some breaking change, or is this just a bug?

--- a/src/uhi/numpy_plottable.py
+++ b/src/uhi/numpy_plottable.py
@@ -142,9 +142,18 @@ class NumPyPlottableHistogram:
 
         raw_bins: tuple[np.typing.NDArray[Any] | None, ...] = bins  # type: ignore[assignment]
 
+        if len(raw_bins) > len(hist.shape):
+            msg = f"Too many bin arrays ({len(raw_bins)}) for histogram with {hist.ndim} dimension(s)"
+            raise ValueError(msg)
+
         self.kind = kind
+        # zip over only the last len(raw_bins) dimensions, so that a 2-D
+        # "stacked" values array (e.g. multiple 1-D histograms sharing one
+        # axis) is handled naturally: the leading dimension(s) are treated as
+        # a stacking dimension and produce no axis entry.
         self.axes: Sequence[PlottableAxis] = [
-            _bin_helper(shape, b) for shape, b in zip(hist.shape, raw_bins, strict=True)
+            _bin_helper(shape, b)
+            for shape, b in zip(hist.shape[-len(raw_bins) :], raw_bins, strict=True)
         ]
 
     def __repr__(self) -> str:

--- a/tests/test_ensure.py
+++ b/tests/test_ensure.py
@@ -91,6 +91,43 @@ def test_from_bh_integer() -> None:
     assert h.axes[0][2] == 3
 
 
+def test_from_numpy_stacked() -> None:
+    """Regression test for issue #233.
+
+    A tuple of (list_of_value_arrays, bin_edges) — a stacked histogram format
+    used by histoprint — was mishandled: the list was converted to a 2D array
+    but only one bin dimension was supplied, causing zip() to raise ValueError.
+    """
+    rng = np.random.default_rng(42)
+    bins = np.linspace(-5, 5, 16)  # 15 bins → 16 edges
+    values = [rng.integers(0, 300, size=15) for _ in range(3)]
+    hist = (values, bins)
+
+    h = ensure_plottable_histogram(hist)
+
+    assert h.values() == approx(np.asarray(values))
+    assert len(h.axes) == 1
+    assert len(h.axes[0]) == 15
+    assert h.axes[0][0] == approx(bins[0:2])
+    assert h.axes[0][-1] == approx(bins[-2:])
+
+
+def test_from_numpy_excess_bins_raises() -> None:
+    """If more bin arrays are provided than histogram dimensions, raise.
+
+    This exercises the new guard that prevents passing e.g. two bin arrays
+    for a 1-D values array.
+    """
+    values = np.arange(5)
+    bins1 = np.linspace(0, 5, 6)
+    bins2 = np.linspace(0, 5, 6)
+    # Provide two bin arrays for a 1-D histogram
+    hist = (values, (bins1, bins2))
+
+    with pytest.raises(ValueError, match="Too many bin arrays"):
+        ensure_plottable_histogram(hist)
+
+
 def test_from_bh_str_cat() -> None:
     bh = pytest.importorskip("boost_histogram")
     h1 = bh.Histogram(bh.axis.StrCategory(["hi", "ho"]))

